src/nnet3/nnet-simple-component.h

```diff
@@ -873,6 +875,83 @@ class PerElementScaleComponent: public UpdatableComponent {
 };
 
 
+// PerElementOffsetComponent offsets each dimension of its input with a separate
+// trainable bias; it's like an affine component with fixed weight matrix which is always equal to I.
+class PerElementOffsetComponent: public UpdatableComponent {
+ public:
+  virtual int32 InputDim() const { return offsets_.Dim(); }
+  virtual int32 OutputDim() const { return offsets_.Dim(); }
+
+  virtual std::string Info() const;
+  virtual void InitFromConfig(ConfigLine *cfl);
+
+  PerElementOffsetComponent() { } // use Init to really initialize.
+  virtual std::string Type() const { return "PerElementOffsetComponent"; }
+  virtual int32 Properties() const {
+    return kSimpleComponent|kUpdatableComponent|
+        kLinearInParameters|kPropagateInPlace;
+  }
+
+  virtual void Propagate(const ComponentPrecomputedIndexes *indexes,
+                         const CuMatrixBase<BaseFloat> &in,
+                         CuMatrixBase<BaseFloat> *out) const;
+  virtual void Backprop(const std::string &debug_info,
+                        const ComponentPrecomputedIndexes *indexes,
+                        const CuMatrixBase<BaseFloat> &in_value,
```

Comment:
change 
<code>const CuMatrixBase<BaseFloat> &in_value</code>
to
<code>const CuMatrixBase<BaseFloat> &</code> as it will be clear to readers that we won't use in_value